src/include/migraphx/operators.hpp

```diff
@@ -1356,6 +1356,108 @@ struct undefined
     argument compute(const shape&, const std::vector<argument>&) const { return {{}, nullptr}; }
 };
 
+struct batch_contiguous
+{
+    int axis = 0;
+    std::vector<int> slice_dims;
+    std::string name() const { return "batch_contiguous"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        auto input_shape = inputs[0];
+        std::vector<std::size_t> out_dims;
+        out_dims.push_back(input_shape.elements());
+        return {input_shape.type(), out_dims};
+    }
+
+    std::vector<uint64_t> compute_index_map(const shape& s) const
+    {
+        std::vector<uint64_t> index_map;
+        uint64_t unit_slice = 1;
+        int axis_id         = 0;
+        if((axis < 0) || (axis >= s.lens().size()))
+            MIGRAPHX_THROW("batch_contiguous: invalid split axis");
+
+        for(auto&& len : s.lens())
+        {
+            if(axis_id++ > axis)
+                unit_slice *= len;
+        }
+        int total_slice_dim = 0;
+        for(auto&& dim : slice_dims)
+        {
+            if(dim == 0)
+                MIGRAPHX_THROW("batch_contiguous: invalid split dimension");
+            total_slice_dim += dim;
+        }
+        if((total_slice_dim == 0) || (total_slice_dim != s.lens()[axis]))
```

Comment:
I am not sure if the first condition (total_slice_dim == 0) is useful since the previous check would ensure all dims are non-zero.

Reply:
You are right.